xgi/classes/reportviews.py

```diff
@@ -237,7 +237,24 @@ def filterby_attr(self, attr, val, mode="eq"):
         <https://github.com/ComplexGroupInteractions/xgi/blob/main/tutorials/Tutorial%206%20-%20Statistics.ipynb>`_.
 
```

Comment:
Notes
--------
Beware of using comparison modes ("lt", "gt", "leq", "geq") when the attribute is a string. For example, the string comparison `'10' < '9'` evaluates to `True`.